lib/sandbox.macos.ml

```diff
@@ -114,6 +114,10 @@ let finished () =
   Os.sudo [ "zfs"; "mount"; "obuilder/result" ] >>= fun () ->
   Lwt.return ()
 
+let shell _ = None
```

Comment:
The context is that all the sandboxes invoke command `x` by running `/usr/bin/env bash -c x` where that default value comes from `lib/build.ml` by passing no parameter for `shell`.  [Hence why my `tar` implementation used the same concept].  Shell is defined as:

```
let shell = Option.value ~default:(if Sys.win32 then ["cmd"; "/S"; "/C"] else ["/usr/bin/env"; "bash"; "-c"]) shell in
```

This only makes sense when `Sys` is the machine that we are building on, but in this case we build on a different architecture to the one we target.  Also since commands are invoked via SSH, they already have a default shell so adding an extra one isn't needed.

I've reworked the original implementation of the default shell to follow the tar implementation...

Check `main.ml` for how it is used and of course `ocluster-worker`